ulib/FStar.Calc.fst

```diff
@@ -0,0 +1,41 @@
+module FStar.Calc
+
+open FStar.Preorder (* for `relation` *)
+
+noeq
```

Comment:
Can this type be `abstract`? We will also have to make `calc_init` and `calc_step` as `abstract`.